### PR TITLE
[4.1] Use the image JLayout for all the images

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request is a continuation of https://github.com/joomla/joomla-cms/pull/37008, https://github.com/joomla/joomla-cms/pull/36613 and https://github.com/joomla/joomla-cms/pull/35780

### Summary of Changes
- Render images using the jLayout

### Testing Instructions
Check the images render as expected

### Documentation Changes Required

I didn't touch the `mod_banners/tmpl/default.php` but that one should also be patched

### CODE REVIEW

**Comment by laoneo** [comment]
Somehow I find it strange that we can't write a simple image tag anymore in core.

**Comment by dgrammatiko** [comment]
> Somehow I find it strange that we can't write a simple image tag anymore in core.


Easily fixable:
- remove remote adapters
- Remove support for the web spec `loading=lazy`

and it would be like 2010 😀

**Comment by laoneo** [comment]
2010 was not bad, less complicated as now :rofl:

**Comment by dgrammatiko** [comment]
> 2010 was not bad, less complicated as now 🤣

I mean you can remove the complicated JLayout and add the same functionality in the sef.php plugin. BUT there's a performance cost, with the JLayout approach you run the code only for images using the media field but with the sef plugin you will run that code for **each and every image in the HTML**. And since for me User Experience is more beneficial than Developer's experience/convenience thus this PR. The project could very easily destroy any meaningful SEO/ROI by executing code repeatedly for cases that is not required but then again will keep devs that think it's awkward to cater themselves for the output of their code very happy. Not up to me to decide, I just proposed what seems **reasonable** by my own standards. Feel free to close this and add the code in the sef.php...

**Comment by bembelimen** [comment]
I still would prefere to implement the layout properly in ```HTMLHelper::image``` and use this everywhere, it's much more convenient

**Comment by dgrammatiko** [comment]
> I still would prefere to implement the layout properly in HTMLHelper::image and use this everywhere, it's much more convenient

Probably I would also agreed with this if the `HTMLHelper::image` had more friendly/memorable arguments:

```php
	/**
	 * Write a `<img>` element
	 *
	 * @param   string        $file        The relative or absolute URL to use for the src attribute.
	 * @param   string        $alt         The alt text.
	 * @param   array|string  $attribs     Attributes to be added to the `<img>` element
	 * @param   boolean       $relative    Flag if the path to the file is relative to the /media folder (and searches in template).
	 * @param   integer       $returnPath  Defines the return value for the method:
	 *                                     -1: Returns a `<img>` tag without looking for relative files
	 *                                     0: Returns a `<img>` tag while searching for relative files
	 *                                     1: Returns the file path to the image while searching for relative files
	 *
	 * @return  string|null  HTML markup for the image, relative path to the image, or null if path is to be returned but image is not found
	 *
	 * @since   1.5
	 */
```

But that last integer it's a very good lesson for new devs **how NOT to assign arguments on a function** 😀

Then again this is just code and someone could maybe fix this in a B/C way (we did that for the script/stylesheet functions somewhere around 3.5). That said I still believe the JLayout is the right API for echoing HTML fragments, it's what the project uses everywhere else (all bootstrap, fields, XTDButtons, etc)

File: components/com_finder/tmpl/search/default_result.php [hunk 1]
```diff
@@ -49,47 +48,39 @@
...
-	$image             = HTMLHelper::cleanImageURL($this->result->imageUrl);
-	$extraAttr         = '';
-
-	// Set lazyloading only for images which have width and height attributes
-	if ((isset($image->attributes['width']) && (int) $image->attributes['width'] > 0)
-		&& (isset($image->attributes['height']) && (int) $image->attributes['height'] > 0))
-	{
-		$extraAttr = ArrayHelper::toString($image->attributes) . ' loading="lazy"';
-	}
+	$extraAttr['class'] = $this->params->get('image_class', '');
```
Comment (Quy) [inline components/com_finder/tmpl/search/default_result.php]:
```suggestion
	$extraAttr['class'] = $imageClass;
```

File: components/com_finder/tmpl/search/default_result.php [hunk 2]
```diff
@@ -49,47 +48,39 @@
 	$description = HTMLHelper::_('string.truncate', StringHelper::substr($full_description, $start), $desc_length, true);
 }
 
-$showImage = $this->params->get('show_image', 0);
+$showImage  = $this->params->get('show_image', 0);
+$imageClass = $this->params->get('image_class', '');
+$extraAttr  = [];
 
-if ($showImage && !empty($this->result->imageUrl))
+if ($showImage && !empty($this->result->imageUrl) && '' !== $imageClass)
```
Comment (Quy) [inline components/com_finder/tmpl/search/default_result.php]:
```suggestion
if ($showImage && !empty($this->result->imageUrl) && $imageClass !== '')
```

File: components/com_finder/tmpl/search/default_result.php [hunk 3]
```diff
@@ -49,47 +48,39 @@
...
 <li class="result__item">
 	<?php if (isset($image)) : ?>
 		<figure class="<?php echo htmlspecialchars($imageclass, ENT_COMPAT, 'UTF-8'); ?> result__image">
 			<?php if ($this->params->get('link_image') && $this->result->route) : ?>
 				<a href="<?php echo Route::_($this->result->route); ?>">
-					<img src="<?php echo htmlspecialchars($image->url, ENT_COMPAT, 'UTF-8'); ?>"
-						 alt="<?php echo htmlspecialchars($this->result->imageAlt, ENT_COMPAT, 'UTF-8'); ?>"
-							<?php echo $extraAttr; ?>
-					/>
+					<?php $extraAttr = array_merge($extraAttr, ['src' => $image->url, 'alt' => $this->result->imageAlt]); ?>
```
Comment (Quy) [inline components/com_finder/tmpl/search/default_result.php]: Line 78 can be moved after line 75 and delete line 82.

Comment (dgrammatiko) [reply components/com_finder/tmpl/search/default_result.php]: I hope I got it right

**Comment by simbus82** [comment]
I have tested this item :white_check_mark: successfully on 444875771193ca7ef890e3c9cb5b216000cd6efa<br /><br />Images are well rendered, i have tested feeds too!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37022">issues.joomla.org/tracker/joomla-cms/37022</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 444875771193ca7ef890e3c9cb5b216000cd6efa<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37022">issues.joomla.org/tracker/joomla-cms/37022</a>.</sub>

**Comment by jwaisner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37022">issues.joomla.org/tracker/joomla-cms/37022</a>.</sub>

**Comment by laoneo** [comment]
Thanks!

**Comment by dgrammatiko** [comment]
Nice, thanks!

**Comment by laoneo** [comment]
Can you also have a look on https://github.com/joomla/joomla-cms/pull/37148?

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on 81a98cfc84bf5fc6bec1aa4288965cb202b6af1b<br /><br />Tested all views<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37022">issues.joomla.org/tracker/joomla-cms/37022</a>.</sub>

**Comment by dgrammatiko** [comment]
@RickR2H this was merged on 25th of February (it's not a pending PR)

**Comment by dgrammatiko** [comment]
Please close this and test #37654

**Pull request merged**